CMakeLists.txt

```diff
@@ -7,14 +7,31 @@ if (NOT CMAKE_VERSION VERSION_LESS 2.8.12)
 endif()
 
 # Allow user to optionally override LLVM build tree locations.
+set(LLVM_BUILD_DIR "" CACHE PATH "Path to LLVM build directory")
 set(LLVM_TOOLS_BINARY_DIR "" CACHE PATH "Path to llvm/bin")
 set(LLVM_LIBRARY_DIR "" CACHE PATH "Path to llvm/lib")
 set(LLVM_MAIN_INCLUDE_DIR "" CACHE PATH "Path to llvm/include")
 set(LLVM_BINARY_DIR "" CACHE PATH "Path to LLVM build tree")
 set(LLVM_MAIN_SRC_DIR "" CACHE PATH "Path to LLVM source tree")
 
-find_package(LLVM REQUIRED NO_CMAKE_PATH) # config mode search
-message(STATUS "Found LLVM: ${LLVM_DIR}")
+if(NOT LLVM_BUILD_DIR)
+  # compatibility with the previous build instructions
+  if(LLVM_DIR)
+    set(LLVM_BUILD_DIR "${LLVM_DIR}/../../..")
+  else()
+    message(FATAL_ERROR "LLVM_BUILD_DIR not defined! It must point to the directory where LLVM was built.")
+  endif()
+endif()
+# only search in ${LLVM_BUILD_DIR} and not in any of the default search locations:
+# this is done by passing PATHS ${LLVM_DIR} and NO_CMAKE_PATH NO_CMAKE_ENVIRONMENT_PATH
+# message("Looking for LLVM in ${LLVM_BUILD_DIR}")
+find_package(LLVM REQUIRED CONFIG PATHS ${LLVM_BUILD_DIR}
+    NO_DEFAULT_PATH NO_CMAKE_ENVIRONMENT_PATH NO_CMAKE_PATH
+    NO_SYSTEM_ENVIRONMENT_PATH NO_CMAKE_PACKAGE_REGISTRY
+    NO_CMAKE_BUILDS_PATH NO_CMAKE_SYSTEM_PATH NO_CMAKE_SYSTEM_PACKAGE_REGISTRY
+    NO_CMAKE_FIND_ROOT_PATH
+)
+message(STATUS "Found LLVM: ${LLVM_BUILD_BINARY_DIR}")
 
```

Comment:
Yes they do, but one is supposed to be passed on the command line when running CMake, the other one is defined in LLVMConfig.cmake. LLVM_BUILD_BINARY_DIR is a bit confusing, from the name I would expect it to point to <build>/bin, but it doesn't.

I'll get rid of the other NO_* flags